File: LeetCode/problems/49-group-anagrams/49-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    let blah = new Map();
    for(let s of strs){
        let key = s.split("").sort().join("");
        if(!blah.has(key)){ blah.set(key,[]); }
        blah.get(key).push(s);
    }
    return [...blah.values()];
};

File: LeetCode/problems/49-group-anagrams/49-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, ArrayList<String>> blah = new HashMap<String, ArrayList<String>>();
        for(String s : strs ){
            char[] arr = s.toCharArray();
            Arrays.sort(arr);
            String key = new String(arr);

            blah.computeIfAbsent(key, k -> new ArrayList<>()).add(s);
        }

        return new ArrayList<>(blah.values());
    }
}
